src/users/users.controller.spec.ts

```diff
@@ -0,0 +1,18 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { UsersController } from './users.controller';
+
+describe('UsersController', () => {
+  let controller: UsersController;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [UsersController],
+    }).compile();
+
+    controller = module.get<UsersController>(UsersController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
```

Comment:
if you tried running `npm run test` script you'll face bloody failure tests in your terminal,
one of the reasons that when you fire the tests, nests inject `UsersController` and tries to resolve all its related dependencies such as [`UsersService`, database..etc]  and since no service provider has been provided here, an error will be thrown. To solve this problem, you can easily add `providers` into line 9, sth like this:
```js
providers: [
      {
        provide: UsersService,
        useValue: {
          findAll: jest.fn()
        }
      }
    ]
 ```
this has to be implemented in both `UsersService` & `UsersContoller` testing files

**Note**: I've tried `providers: [UsersService]` and it didn't work out. test it yourself
and please make sure to pass all the tests so **CircleCI** can rest in hell :fire: